sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/SessionTokenHelper.java

```diff
@@ -60,6 +58,12 @@ public static void setPartitionLocalSessionToken(RxDocumentServiceRequest reques
         }
     }
 
+    public static void setPartitionLocalSessionToken(RxDocumentServiceRequest request,
+                                                     ISessionContainer sessionContainer) {
+        setPartitionLocalSessionToken(request, request.requestContext.resolvedPartitionKeyRange.getId(),
```

Comment:
I've observed cases in my local testing where requestContext.resolvedPartitionKeyRange is actually "null".

Reply:
As said in another comment https://github.com/Azure/azure-sdk-for-java/pull/25555#discussion_r754451241 , this is covered in test cases. Please check the complete logic